src/test/test-capability.c

```diff
@@ -77,6 +86,34 @@ static int setup_tests(void) {
         test_uid = nobody->pw_uid;
         test_gid = nobody->pw_gid;
 
+        /* Find out the kernel version to determine if the ambient
+         * capability tests should be run. */
+
+        if (uname(&u) < 0)
+                return -errno;
+
+        version = strdup(u.release);
+        orig = version;
+
+        token = strsep(&version, ".");
+        if (!token || !version) {
+                return -EINVAL;
+        }
+
+        r = safe_atou(token, &major);
+        if (r < 0)
+                return r;
+
+        /* Can't use safe_atou here because the string is not clean. */
+        minor = strtoul(version, &end, 10);
+        if (version == end) {
+                return -EINVAL;
+        }
+
+        /* Only run ambient capability tests on kernel versions >= 4.3. */
+        if (major > 4 || (major == 4 && minor >= 3))
+                *run_ambient = true;
+
```

Comment:
Yes, we should definitely check for the actual feature we want to use, not for the version number and make assumption on how we expect the kernel to behave. We can give the new implementation a good test on different kernels, and the fallback path must be defensive (ie, gracefully downgrade the functionality if we believe ambient caps are unavailable).